src/stream.rs

```diff
@@ -367,47 +391,73 @@ pub fn compress_buf(params: &CompressParams,
 pub fn compress_vec(params: &CompressParams,
```

Comment:
Actually, it seems like it's the other way round.  `BrotliEncoderCompressStream` internally calls `BrotliEncoderWriteData`.

I guess if anything, I should change `compress_vec` to use the `Compress` functions.  That should make it much simpler and will allow the removal of another `unsafe` block.